src/pages/EntityDetails/Model/ApplicationsTab/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,75 @@
+import Fuse from "fuse.js";
+import { useState, useEffect, useMemo } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useParams } from "react-router-dom";
+
+import type { EntityDetailsRoute } from "components/Routes/Routes";
+import { useQueryParams } from "hooks/useQueryParams";
+import type { ApplicationData, ApplicationInfo } from "juju/types";
+import { actions as jujuActions } from "store/juju";
+import {
+  getModelApplications,
+  getModelUUIDFromList,
+} from "store/juju/selectors";
+
+import LocalAppsTable from "../LocalAppsTable";
+
+const SearchResults = () => {
+  const dispatch = useDispatch();
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const modelUUID = useSelector(getModelUUIDFromList(modelName, userName));
+  const applications = useSelector(getModelApplications(modelUUID));
+  const [filteredApplications, setFilteredApplications] =
+    useState<ApplicationData>({});
+  const [fuse, setFuse] = useState<Fuse<ApplicationInfo>>(new Fuse([]));
+  const [{ filterQuery }] = useQueryParams<{
+    filterQuery: string;
+  }>({
+    filterQuery: "",
+  });
+
+  useMemo(() => {
+    if (applications)
+      setFuse(
+        new Fuse(Object.values(applications), {
+          keys: ["name", "charm-url", "owner-tag", "constraints.arch"],
+        })
+      );
+  }, [applications]);
+
+  useEffect(() => {
+    if (!applications) {
+      return;
+    }
```

Comment:
I'm not sure how important this is, but the following conditional statement from the initial code in `ApplicationsTab` component is missing:

```
if (!filterQuery) {
  setFilteredApplications(applications);
}
```

Reply:
Oh I think I removed that because now that this logic exists in the SearchResults component it is only displayed when there is a `filterQuery`: https://github.com/canonical/juju-dashboard/pull/1527/files#diff-26406d98c6af45bbb692232a87a220f3913d19c43ceae25855157a1d37d1db5dR176